src/main/java/org/reaktivity/nukleus/http2/internal/types/Http2DecodeDataFW.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2016-2020 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.http2.internal.types;
+
+import static org.reaktivity.nukleus.http2.internal.stream.Http2Flags.END_STREAM;
+import static org.reaktivity.nukleus.http2.internal.types.Http2FrameType.DATA;
+
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.reaktivity.nukleus.http2.internal.stream.Http2Flags;
+
+/*
+    Flyweight for HTTP2 DATA frame
+
+    +-----------------------------------------------+
+    |                 Length (24)                   |
+    +---------------+---------------+---------------+
+    |   Type (8)    |   Flags (8)   |
+    +-+-------------+---------------+-------------------------------+
+    |R|                 Stream Identifier (31)                      |
+    +=+=============+===============================================+
+    |Pad Length? (8)|
+    +---------------+-----------------------------------------------+
+    |                            Data (*)                         ...
+    +---------------------------------------------------------------+
+    |                           Padding (*)                       ...
+    +---------------------------------------------------------------+
+
+ */
+public class Http2DecodeDataFW extends Http2FrameFW
```

Comment:
Why is this new file needed? Should we not just modify the existing `Http2DataFW` instead?

Reply:
That's what I did initially but then I realized that we need that to encode data with the payload. Makes sense?

Reply:
Makes sense